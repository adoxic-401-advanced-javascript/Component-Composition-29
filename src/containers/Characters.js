import React, { Component } from 'react';
import PropType from 'prop-types';
import DeckOfCards from '../components/DeckOfCards';

export default class Characters extends Component {
  static propTypes = {
    count: PropType.number
  }

  static defaultProps = {
    count: 20
  }

  state = {
    characters: []
  }

  render() {
    return (
      <DeckOfCards 
        items={this.state.characters}
        title="Characters"/>
    );
  }

  componentDidMount() {
    this.apiCall();
  }

  apiCall = () => {
    return fetch(`https://hey-arnold-api.herokuapp.com/api/v1/characters?perPage=${this.props.count}`)
      .then(res => res.json())
      .then(characters => {
        this.setState({ characters }); 
      });

  }
}
