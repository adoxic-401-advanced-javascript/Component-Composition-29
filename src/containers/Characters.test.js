import React from 'react';
import { shallow, mount } from 'enzyme';
import Characters from './Characters';



describe('Characters container', () => {

  it('renders Character component', () => {

    global.fetch = jest.fn(() => Promise.resolve());
    const wrapper = shallow(<Characters />);
    expect(wrapper).toMatchSnapshot();
  });

});
