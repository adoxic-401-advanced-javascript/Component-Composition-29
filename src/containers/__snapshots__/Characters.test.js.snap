// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters container renders Character component 1`] = `
<DeckOfCards
  items={Array []}
  title="Characters"
/>
`;
